<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using the rrvgo package • rrvgo</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using the rrvgo package">
<meta property="og:description" content="rrvgo">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rrvgo</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.15.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/rrvgo.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Using the rrvgo package</h1>
                        <h4 data-toc-skip class="author">Sergi
Sayols</h4>
            
            <h4 data-toc-skip class="date">2023-12-15</h4>
      
      
      <div class="hidden name"><code>rrvgo.Rmd</code></div>

    </div>

    
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div class="section level2">
<h2 id="introduction-to-rrvgo">Introduction to rrvgo<a class="anchor" aria-label="anchor" href="#introduction-to-rrvgo"></a>
</h2>
<p>Gene Ontologies (GO) are often used to guide the interpretation of
high-throughput omics experiments, with lists of differentially
regulated genes being summarized into sets of genes with a common
functional representation. Due to the hierachical nature of Gene
Ontologies, the resulting lists of enriched sets are usually redundant
and difficult to interpret.</p>
<p><code>rrvgo</code> aims at simplifying the redundance of GO sets by
grouping similar terms based on their semantic similarity. It also
provides some plots to help with interpreting the summarized terms.</p>
<p>This software is heavily influenced by <a href="http://revigo.irb.hr/" class="external-link">REVIGO</a>. It mimics a good part of its
core functionality, and even some of the outputs are similar. Without
aims to compete, <code>rrvgo</code> tries to offer a programatic
interface using available annotation databases and semantic similarity
methods implemented in the Bioconductor project.</p>
</div>
<div class="section level2">
<h2 id="using-rrvgo">Using rrvgo<a class="anchor" aria-label="anchor" href="#using-rrvgo"></a>
</h2>
<div class="section level3">
<h3 id="getting-started">Getting started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h3>
<p>Starting with a list of genes of interest (eg. coming from a
differential expression analysis), apply any method for the
identification of eneriched GO terms (see <a href="https://www.bioconductor.org/packages/release/bioc/html/GOstats.html" class="external-link">GOStats</a>
or <a href="https://www.gsea-msigdb.org/gsea/index.jsp%5D" class="external-link">GSEA</a>).</p>
<p><code>rrvgo</code> does not care about genes, but GO terms. The input
is a vector of enriched GO terms, along with (recommended, but not
mandatory) a vector of scores. If scores are not provided,
<code>rrvgo</code> takes the GO term (set) size as a score, thus
favoring <em>broader</em> terms.</p>
</div>
<div class="section level3">
<h3 id="calculating-the-similarity-matrix-and-reducing-go-terms">Calculating the similarity matrix and reducing GO terms<a class="anchor" aria-label="anchor" href="#calculating-the-similarity-matrix-and-reducing-go-terms"></a>
</h3>
<p>First step is to get the similarity matrix between terms. The
function <code>calculateSimMatrix</code> takes a list of GO terms for
which the semantic simlarity is to be calculated, an <code>OrgDb</code>
object for an organism, the ontology of interest and the method to
calculate the similarity scores.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://www.bioconductor.org/packages/rrvgo" class="external-link">rrvgo</a></span><span class="op">)</span></span>
<span><span class="va">go_analysis</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/example.txt"</span>, package<span class="op">=</span><span class="st">"rrvgo"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">simMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateSimMatrix.html">calculateSimMatrix</a></span><span class="op">(</span><span class="va">go_analysis</span><span class="op">$</span><span class="va">ID</span>,</span>
<span>                                orgdb<span class="op">=</span><span class="st">"org.Hs.eg.db"</span>,</span>
<span>                                ont<span class="op">=</span><span class="st">"BP"</span>,</span>
<span>                                method<span class="op">=</span><span class="st">"Rel"</span><span class="op">)</span></span></code></pre></div>
<p>The <code>semdata</code> parameter (see
<code><a href="../reference/calculateSimMatrix.html">?calculateSimMatrix</a></code>) is not mandatory as it is calculated
on demand. If the function needs to run several times with the same
organism, it’s advisable to save the
<code>GOSemSim::godata(orgdb, ont=ont)</code> object, in order to reuse
it between calls and speedup the calculation of the similarity
matrix.</p>
<p>From the similarity matrix one can group terms based on similarity.
<code>rrvgo</code> provides the <code>reduceSimMatrix</code> function
for that. It takes as arguments i) the similarity matrix, ii) an
optional <em>named</em> vector of scores associated to each GO term,
iii) a similarity threshold used for grouping terms, and iv) an orgdb
object.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">go_analysis</span><span class="op">$</span><span class="va">qvalue</span><span class="op">)</span>, <span class="va">go_analysis</span><span class="op">$</span><span class="va">ID</span><span class="op">)</span></span>
<span><span class="va">reducedTerms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/reduceSimMatrix.html">reduceSimMatrix</a></span><span class="op">(</span><span class="va">simMatrix</span>,</span>
<span>                                <span class="va">scores</span>,</span>
<span>                                threshold<span class="op">=</span><span class="fl">0.7</span>,</span>
<span>                                orgdb<span class="op">=</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span></code></pre></div>
<p><code>reduceSimMatrix</code> groups terms which are at least within a
similarity below <code>threshold</code>, and selects as the group
representative the term with the higher score within the group. In case
the vector of scores is not available, <code>reduceSimMatrix</code> can
either use the <em>uniqueness</em> of a term (default), or the GO term
<em>size</em>. In the case of <em>size</em>, <code>rrvgo</code> will
fetch the GO term size from the <code>OrgDb</code> object and use it as
the score, thus favoring broader terms. <strong>Please note that scores
are interpreted in the direction that higher are better</strong>,
therefore if you use p-values as scores, minus log-transform them
before.</p>
<p><strong>NOTE:</strong><code>rrvgo</code> uses the similarity between
pairs of terms to compute a distance matrix, defined as
<code>(1-simMatrix)</code>. The terms are then hierarchically clustered
using complete linkage, and the tree is cut at the desired threshold,
picking the term with the highest score as the representative of each
group.<br>
Therefore, higher thresholds lead to fewer groups, and the threshold
should be read as the minimum similarity between group
representatives.</p>
</div>
<div class="section level3">
<h3 id="plotting-and-interpretation">Plotting and interpretation<a class="anchor" aria-label="anchor" href="#plotting-and-interpretation"></a>
</h3>
<p><code>rrvgo</code> provides several methods for plotting and
interpreting the results.</p>
<div class="section level4">
<h4 id="similarity-matrix-heatmap">Similarity matrix heatmap<a class="anchor" aria-label="anchor" href="#similarity-matrix-heatmap"></a>
</h4>
<p>Plot similarity matrix as a heatmap, with clustering of columns of
rows turned on by default (thus arranging together similar terms).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/heatmapPlot.html">heatmapPlot</a></span><span class="op">(</span><span class="va">simMatrix</span>,</span>
<span>            <span class="va">reducedTerms</span>,</span>
<span>            annotateParent<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>            annotationLabel<span class="op">=</span><span class="st">"parentTerm"</span>,</span>
<span>            fontsize<span class="op">=</span><span class="fl">6</span><span class="op">)</span></span></code></pre></div>
<p><img src="rrvgo_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
<p>The function internally uses <a href="https://cran.r-project.org/web/packages/pheatmap/index.html" class="external-link"><code>pheatmap</code></a>,
and further parameters can be passed to this function.</p>
</div>
<div class="section level4">
<h4 id="scatter-plot-depicting-groups-and-distance-between-terms">Scatter plot depicting groups and distance between terms<a class="anchor" aria-label="anchor" href="#scatter-plot-depicting-groups-and-distance-between-terms"></a>
</h4>
<p>Plot GO terms as scattered points. Distances between points represent
the similarity between terms, and axes are the first 2 components of
applying a PCoA to the (di)similarity matrix. Size of the point
represents the provided scores or, in its absence, the number of genes
the GO term contains.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">simMatrix</span>, <span class="va">reducedTerms</span><span class="op">)</span></span></code></pre></div>
<p><img src="rrvgo_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="treemap-plot">Treemap plot<a class="anchor" aria-label="anchor" href="#treemap-plot"></a>
</h4>
<p>Treemaps are space-filling visualization of hierarchical structures.
The terms are grouped (colored) based on their parent, and the space
used by the term is proportional to the score. Treemaps can help with
the interpretation of the summarized results and also comparing
differents sets of GO terms.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/treemapPlot.html">treemapPlot</a></span><span class="op">(</span><span class="va">reducedTerms</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="figures/treemap.png" alt=""><p class="caption">treemap</p>
</div>
<p>The function internally uses <a href="https://cran.r-project.org/web/packages/treemap/index.html" class="external-link"><code>treemap</code></a>,
and further parameters can be passed to this function.</p>
</div>
<div class="section level4">
<h4 id="word-cloud">Word cloud<a class="anchor" aria-label="anchor" href="#word-cloud"></a>
</h4>
<p>Word clouds are visualizations which reproduce a text putting
emphasis to words which appear frequently in a text. They can help to
identify processes and functions that happen more commonly in a set of
enriched GO terms, as well as comparing between different sets.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/wordcloudPlot.html">wordcloudPlot</a></span><span class="op">(</span><span class="va">reducedTerms</span>, min.freq<span class="op">=</span><span class="fl">1</span>, colors<span class="op">=</span><span class="st">"black"</span><span class="op">)</span></span></code></pre></div>
<p><img src="rrvgo_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>The function internally uses <a href="https://cran.r-project.org/web/packages/wordcloud/index.html" class="external-link"><code>wrodcloud</code></a>,
and further parameters can be passed to this function.</p>
</div>
</div>
<div class="section level3">
<h3 id="shiny-app">Shiny app<a class="anchor" aria-label="anchor" href="#shiny-app"></a>
</h3>
<p>To make the software more accessible to a non-technical audience,
<code>rrvgo</code> packages a shiny app which can be accessed calling
the <code><a href="../reference/shiny_rrvgo.html">shiny_rrvgo()</a></code> function from the R console.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rrvgo</span><span class="fu">::</span><span class="fu"><a href="../reference/shiny_rrvgo.html">shiny_rrvgo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="figures/shiny_app.png" alt=""><p class="caption">shiny_app</p>
</div>
<p>The app offers <em>interactive</em> access to the plots and tables
calculated by <code>rrvgo</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="currently-supported">Currently supported<a class="anchor" aria-label="anchor" href="#currently-supported"></a>
</h2>
<div class="section level3">
<h3 id="similarity-methods">Similarity methods<a class="anchor" aria-label="anchor" href="#similarity-methods"></a>
</h3>
<p>All similarity measures available are those implemented in the <a href="https://www.bioconductor.org/packages/release/bioc/html/GOSemSim.html" class="external-link">GOSemSim
package</a>, namely the Resnik, Lin, Relevance, Jiang and Wang methods.
See the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/GOSemSim/inst/doc/GOSemSim.html#semantic-similarity-measurement-based-on-go" class="external-link">Semantic
Similarity Measurement Based on GO</a> section from the GOSeSim
documentation for more details.</p>
</div>
<div class="section level3">
<h3 id="organisms">Organisms<a class="anchor" aria-label="anchor" href="#organisms"></a>
</h3>
<p>Bioconductor current provides <code>OrgDb</code> objects for <a href="https://www.bioconductor.org/packages/release/BiocViews.html#___OrgDb" class="external-link">20
species</a> provided by the following packages:</p>
<table class="table">
<thead><tr class="header">
<th>Package</th>
<th>Organism</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>org.Ag.eg.db</td>
<td>Anopheles</td>
</tr>
<tr class="even">
<td>org.At.tair.db</td>
<td>Arabidopsis</td>
</tr>
<tr class="odd">
<td>org.Bt.eg.db</td>
<td>Bovine</td>
</tr>
<tr class="even">
<td>org.Ce.eg.db</td>
<td>Worm</td>
</tr>
<tr class="odd">
<td>org.Cf.eg.db</td>
<td>Canine</td>
</tr>
<tr class="even">
<td>org.Dm.eg.db</td>
<td>Fly</td>
</tr>
<tr class="odd">
<td>org.Dr.eg.db</td>
<td>Zebrafish</td>
</tr>
<tr class="even">
<td>org.EcK12.eg.db</td>
<td>E coli strain K12</td>
</tr>
<tr class="odd">
<td>org.EcSakai.eg.db</td>
<td>E coli strain Sakai</td>
</tr>
<tr class="even">
<td>org.Gg.eg.db</td>
<td>Chicken</td>
</tr>
<tr class="odd">
<td>org.Hs.eg.db</td>
<td>Human</td>
</tr>
<tr class="even">
<td>org.Mm.eg.db</td>
<td>Mouse</td>
</tr>
<tr class="odd">
<td>org.Mmu.eg.db</td>
<td>Rhesus</td>
</tr>
<tr class="even">
<td>org.Mxanthus.db</td>
<td>Myxococcus xanthus DK 1622</td>
</tr>
<tr class="odd">
<td>org.Pf.plasmo.db</td>
<td>Malaria</td>
</tr>
<tr class="even">
<td>org.Pt.eg.db</td>
<td>Chimp</td>
</tr>
<tr class="odd">
<td>org.Rn.eg.db</td>
<td>Rat</td>
</tr>
<tr class="even">
<td>org.Sc.sgd.db</td>
<td>Yeast</td>
</tr>
<tr class="odd">
<td>org.Ss.eg.db</td>
<td>Pig</td>
</tr>
<tr class="even">
<td>org.Xl.eg.db</td>
<td>Xenopus</td>
</tr>
</tbody>
</table>
<p>If the organism is not supported in Bioconductor, you can still build
your own <code>OrgDb</code> object usign the <a href="https://bioconductor.org/packages/release/bioc/html/AnnotationForge.html" class="external-link"><code>AnnotationForge</code></a>
package and rendering the necessary data for semantic similarity using
the <code>GOSemSim</code> package with:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">my_new_fancy_orgdb_object</span> <span class="op">&lt;-</span> <span class="st">'org.Zz.eg.db'</span></span>
<span><span class="va">hsGO</span> <span class="op">&lt;-</span> <span class="fu">GOSemSim</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GOSemSim/man/godata.html" class="external-link">godata</a></span><span class="op">(</span><span class="va">my_new_fancy_orgdb_object</span>, ont<span class="op">=</span><span class="st">"MF"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="gene-ontologies">Gene Ontologies<a class="anchor" aria-label="anchor" href="#gene-ontologies"></a>
</h3>
<p>One of <em>Biologiocal Process</em> (BP), <em>Molecular Function</em>
(MF) or <em>Cellular Compartment</em> (CC).</p>
</div>
</div>
<div class="section level2">
<h2 id="demo-data">Demo data<a class="anchor" aria-label="anchor" href="#demo-data"></a>
</h2>
<p>Taken as is from the <a href="https://www.bioconductor.org/packages/release/bioc/html/DOSE.html" class="external-link">DOSE
package</a>, which was derived from the R package <a href="https://www.bioconductor.org/packages/release/bioc/html/breastCancerMAINZ.html" class="external-link">breastCancerMAINZ</a>.
It contains 200 samples with breast cancer at different grades (I, II
and III). The dataset basically contains log2 ratios of the geometric
means of grade III vs. grade I samples ( 34 vs. 29 repectively).</p>
</div>
<div class="section level2">
<h2 id="citing-rrvgo">Citing rrvgo<a class="anchor" aria-label="anchor" href="#citing-rrvgo"></a>
</h2>
<p>Please consider citing rrvgo if used in support of your own
research:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"rrvgo"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## To cite package 'rrvgo' in publications use:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Sayols, S (2023). rrvgo: a Bioconductor package for interpreting</span></span>
<span><span class="co">##   lists of Gene Ontology terms. microPublication Biology.</span></span>
<span><span class="co">##   10.17912/micropub.biology.000811</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   @Article{,</span></span>
<span><span class="co">##     title = {rrvgo: a Bioconductor package to reduce and visualize Gene Ontology terms},</span></span>
<span><span class="co">##     author = {Sergi Sayols},</span></span>
<span><span class="co">##     year = {2023},</span></span>
<span><span class="co">##     journal = {microPublication Biology},</span></span>
<span><span class="co">##     doi = {10.17912/micropub.biology.000811},</span></span>
<span><span class="co">##     url = {https://www.micropublication.org/journals/biology/micropub-biology-000811},</span></span>
<span><span class="co">##   }</span></span></code></pre>
<div class="section level3">
<h3 id="reporting-problems-or-bugs">Reporting problems or bugs<a class="anchor" aria-label="anchor" href="#reporting-problems-or-bugs"></a>
</h3>
<p>If you run into problems using rrvgo, the <a href="https://support.bioconductor.org/" class="external-link">Bioconductor Support site</a>
is a good first place to ask for help. If you think there is a bug or an
unreported feature, you can report it using the <a href="https://github.com/ssayols/rrvgo/" class="external-link">rrvgo github site</a>.</p>
</div>
<div class="section level3">
<h3 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<p>The following package and versions were used in the production of
this vignette.</p>
<pre><code><span><span class="co">## R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Debian GNU/Linux trixie/sid</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/atlas/libblas.so.3.10.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/atlas/liblapack.so.3.10.3;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Europe/Madrid</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] rrvgo_1.15.1     knitr_1.45       BiocStyle_2.28.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] DBI_1.1.3               bitops_1.0-7            rlang_1.1.2            </span></span>
<span><span class="co">##   [4] magrittr_2.0.3          gridBase_0.4-7          compiler_4.3.2         </span></span>
<span><span class="co">##   [7] RSQLite_2.3.1           png_0.1-8               systemfonts_1.0.4      </span></span>
<span><span class="co">##  [10] vctrs_0.6.4             stringr_1.5.1           pkgconfig_2.0.3        </span></span>
<span><span class="co">##  [13] crayon_1.5.2            fastmap_1.1.1           XVector_0.40.0         </span></span>
<span><span class="co">##  [16] ellipsis_0.3.2          labeling_0.4.3          utf8_1.2.4             </span></span>
<span><span class="co">##  [19] promises_1.2.1          rmarkdown_2.25          ragg_1.2.5             </span></span>
<span><span class="co">##  [22] purrr_1.0.2             bit_4.0.5               xfun_0.41              </span></span>
<span><span class="co">##  [25] zlibbioc_1.46.0         cachem_1.0.8            GenomeInfoDb_1.36.4    </span></span>
<span><span class="co">##  [28] jsonlite_1.8.8          blob_1.2.4              highr_0.10             </span></span>
<span><span class="co">##  [31] later_1.3.1             parallel_4.3.2          R6_2.5.1               </span></span>
<span><span class="co">##  [34] bslib_0.6.0             stringi_1.8.1           RColorBrewer_1.1-3     </span></span>
<span><span class="co">##  [37] reticulate_1.34.0       jquerylib_0.1.4         GOSemSim_2.26.1        </span></span>
<span><span class="co">##  [40] Rcpp_1.0.11             bookdown_0.36.2         IRanges_2.34.1         </span></span>
<span><span class="co">##  [43] httpuv_1.6.12           Matrix_1.6-4            igraph_1.5.1           </span></span>
<span><span class="co">##  [46] tidyselect_1.2.0        yaml_2.3.7              codetools_0.2-19       </span></span>
<span><span class="co">##  [49] lattice_0.22-5          tibble_3.2.1            Biobase_2.60.0         </span></span>
<span><span class="co">##  [52] shiny_1.8.0             withr_2.5.2             KEGGREST_1.40.1        </span></span>
<span><span class="co">##  [55] askpass_1.2.0           evaluate_0.23           treemap_2.4-4          </span></span>
<span><span class="co">##  [58] desc_1.4.2              xml2_1.3.5              Biostrings_2.68.1      </span></span>
<span><span class="co">##  [61] pillar_1.9.0            BiocManager_1.30.22     stats4_4.3.2           </span></span>
<span><span class="co">##  [64] NLP_0.2-1               generics_0.1.3          rprojroot_2.0.3        </span></span>
<span><span class="co">##  [67] RCurl_1.98-1.13         S4Vectors_0.38.2        ggplot2_3.4.4          </span></span>
<span><span class="co">##  [70] munsell_0.5.0           scales_1.2.1            xtable_1.8-4           </span></span>
<span><span class="co">##  [73] glue_1.6.2              slam_0.1-50             pheatmap_1.0.12        </span></span>
<span><span class="co">##  [76] tools_4.3.2             tm_0.7-11               data.table_1.14.8      </span></span>
<span><span class="co">##  [79] RSpectra_0.16-1         fs_1.6.3                grid_4.3.2             </span></span>
<span><span class="co">##  [82] umap_0.2.10.0           AnnotationDbi_1.62.2    colorspace_2.1-0       </span></span>
<span><span class="co">##  [85] GenomeInfoDbData_1.2.10 cli_3.6.1               textshaping_0.3.6      </span></span>
<span><span class="co">##  [88] fansi_1.0.5             dplyr_1.1.3             gtable_0.3.4           </span></span>
<span><span class="co">##  [91] sass_0.4.7              digest_0.6.33           BiocGenerics_0.46.0    </span></span>
<span><span class="co">##  [94] wordcloud_2.6           ggrepel_0.9.4           org.Hs.eg.db_3.17.0    </span></span>
<span><span class="co">##  [97] farver_2.1.1            memoise_2.0.1           htmltools_0.5.7        </span></span>
<span><span class="co">## [100] pkgdown_2.0.7           lifecycle_1.0.4         httr_1.4.7             </span></span>
<span><span class="co">## [103] GO.db_3.17.0            mime_0.12               openssl_2.1.1          </span></span>
<span><span class="co">## [106] bit64_4.0.5</span></span></code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Sergi Sayols.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
